::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Guillermo Herrero <guillermo.herrero@enea.com>
Date: Wed, 13 Sep 2017 14:15:07 +0200
Subject: [PATCH] Pod Descriptor File compatible reclass structure

Implementation for baremetal-mcp-ocata-ovs-ha scenario

JIRA: FUEL-275

Change-Id: Id6ab5697f993ac9faa019c3c10ba4ed4b7b6db01
Signed-off-by: Guillermo Herrero <guillermo.herrero@enea.com>
---
 .../cluster/all-mcp-ocata-common/opnfv/init.yml    |  1 +
 .../all-mcp-ocata-common/opnfv/pod_config.yml      | 92 ++++++++++++++++++++++
 .../opnfv/runtime.yml.template                     | 10 +--
 .../baremetal-mcp-ocata-odl-ha/infra/init.yml      |  2 +-
 .../baremetal-mcp-ocata-odl-ha/infra/maas.yml      | 10 +--
 .../baremetal-mcp-ocata-ovs-dpdk-ha/infra/init.yml |  2 +-
 .../baremetal-mcp-ocata-ovs-dpdk-ha/infra/maas.yml | 10 +--
 .../baremetal-mcp-ocata-ovs-ha/infra/config.yml    |  1 -
 .../baremetal-mcp-ocata-ovs-ha/infra/init.yml      | 24 +++---
 .../baremetal-mcp-ocata-ovs-ha/infra/kvm.yml       | 11 ++-
 .../baremetal-mcp-ocata-ovs-ha/infra/maas.yml      | 72 ++++++++---------
 .../openstack/compute.yml                          | 12 +--
 .../baremetal-mcp-ocata-ovs-ha/openstack/init.yml  | 72 ++++++++---------
 13 files changed, 205 insertions(+), 114 deletions(-)
 create mode 100644 mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pod_config.yml

diff --git a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/init.yml b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/init.yml
index 28619ba..4cb4f2f 100644
--- a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/init.yml
+++ b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/init.yml
@@ -1,2 +1,3 @@
 classes:
 - cluster.all-mcp-ocata-common.opnfv.runtime
+- cluster.all-mcp-ocata-common.opnfv.pod_config
diff --git a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pod_config.yml b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pod_config.yml
new file mode 100644
index 0000000..368e60f
--- /dev/null
+++ b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pod_config.yml
@@ -0,0 +1,92 @@
+parameters:
+  _param:
+    # infra service addresses
+    opnfv_infra_config_address: 10.167.4.100
+    opnfv_infra_config_deploy_address: 10.20.0.2
+    opnfv_infra_maas_node01_address: 10.167.4.2
+    opnfv_infra_maas_node01_deploy_address: 192.168.11.2
+    opnfv_infra_maas_node01_external_address: 172.30.10.2
+    opnfv_infra_compute_node01_address: 10.167.4.141
+    opnfv_infra_compute_node02_address: 10.167.4.142
+    opnfv_infra_compute_node03_address: 10.167.4.143
+    opnfv_infra_kvm_address: 10.167.4.140
+    opnfv_infra_kvm_node01_address: 10.167.4.141
+    opnfv_infra_kvm_node02_address: 10.167.4.142
+    opnfv_infra_kvm_node03_address: 10.167.4.143
+
+    opnfv_openstack_gateway_node01_address: 10.167.4.124
+    opnfv_openstack_gateway_node02_address: 10.167.4.125
+    opnfv_openstack_gateway_node03_address: 10.167.4.126
+    opnfv_openstack_gateway_node01_tenant_address: 10.1.0.6
+    opnfv_openstack_gateway_node02_tenant_address: 10.1.0.7
+    opnfv_openstack_gateway_node03_tenant_address: 10.1.0.9
+    opnfv_openstack_proxy_address: 10.167.4.80
+    opnfv_openstack_proxy_node01_address: 10.167.4.81
+    opnfv_openstack_proxy_node02_address: 10.167.4.82
+    opnfv_openstack_control_address: 10.167.4.10
+    opnfv_openstack_control_node01_address: 10.167.4.11
+    opnfv_openstack_control_node02_address: 10.167.4.12
+    opnfv_openstack_control_node03_address: 10.167.4.13
+    opnfv_openstack_database_address: 10.167.4.50
+    opnfv_openstack_database_node01_address: 10.167.4.51
+    opnfv_openstack_database_node02_address: 10.167.4.52
+    opnfv_openstack_database_node03_address: 10.167.4.53
+    opnfv_openstack_message_queue_address: 10.167.4.40
+    opnfv_openstack_message_queue_node01_address: 10.167.4.41
+    opnfv_openstack_message_queue_node02_address: 10.167.4.42
+    opnfv_openstack_message_queue_node03_address: 10.167.4.43
+    opnfv_openstack_telemetry_address: 10.167.4.75
+    opnfv_openstack_telemetry_node01_address: 10.167.4.76
+    opnfv_openstack_telemetry_node02_address: 10.167.4.77
+    opnfv_openstack_telemetry_node03_address: 10.167.4.78
+    opnfv_openstack_compute_node01_single_address: 10.167.4.101
+    opnfv_openstack_compute_node02_single_address: 10.167.4.102
+    opnfv_openstack_compute_node03_single_address: 10.167.4.103
+    opnfv_openstack_compute_node01_control_address: 10.167.4.101
+    opnfv_openstack_compute_node02_control_address: 10.167.4.102
+    opnfv_openstack_compute_node03_control_address: 10.167.4.103
+    opnfv_openstack_compute_node01_tenant_address: 10.1.0.101
+    opnfv_openstack_compute_node02_tenant_address: 10.1.0.102
+    opnfv_openstack_compute_node03_tenant_address: 10.1.0.103
+    opnfv_openstack_compute_node01_external_address: 172.30.10.101
+    opnfv_openstack_compute_node02_external_address: 172.30.10.102
+
+
+    opnfv_name_servers: ['8.8.8.8', '8.8.4.4']
+    opnfv_net_mgmt_vlan: 300
+    opnfv_net_tenant_vlan: 302
+
+    opnfv_maas_node01_architecture: amd64/generic
+    opnfv_maas_node01_power_address: 172.30.8.75
+    opnfv_maas_node01_power_type: ipmi
+    opnfv_maas_node01_power_user: admin
+    opnfv_maas_node01_power_password: octopus
+    opnfv_maas_node01_interface_mac: "00:25:b5:a0:00:2a"
+
+    opnfv_maas_node02_architecture: amd64/generic
+    opnfv_maas_node02_power_address: 172.30.8.65
+    opnfv_maas_node02_power_type: ipmi
+    opnfv_maas_node02_power_user: admin
+    opnfv_maas_node02_power_password: octopus
+    opnfv_maas_node02_interface_mac: "00:25:b5:a0:00:3a"
+
+    opnfv_maas_node03_architecture: amd64/generic
+    opnfv_maas_node03_power_address: 172.30.8.74
+    opnfv_maas_node03_power_type: ipmi
+    opnfv_maas_node03_power_user: admin
+    opnfv_maas_node03_power_password: octopus
+    opnfv_maas_node03_interface_mac: "00:25:b5:a0:00:4a"
+
+    opnfv_maas_node04_architecture: amd64/generic
+    opnfv_maas_node04_power_address: 172.30.8.73
+    opnfv_maas_node04_power_type: ipmi
+    opnfv_maas_node04_power_user: admin
+    opnfv_maas_node04_power_password: octopus
+    opnfv_maas_node04_interface_mac: "00:25:b5:a0:00:5a"
+
+    opnfv_maas_node05_architecture: amd64/generic
+    opnfv_maas_node05_power_address: 172.30.8.72
+    opnfv_maas_node05_power_type: ipmi
+    opnfv_maas_node05_power_user: admin
+    opnfv_maas_node05_power_password: octopus
+    opnfv_maas_node05_interface_mac: "00:25:b5:a0:00:6a"
diff --git a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/runtime.yml.template b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/runtime.yml.template
index 121f706..b07ae13 100644
--- a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/runtime.yml.template
+++ b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/runtime.yml.template
@@ -2,8 +2,8 @@ parameters:
   _param:
     reclass_config_master: ${SALT_MASTER}

-    opnfv_fuel_maas_mcp_address: ${MAAS_IP}
-    opnfv_fuel_maas_pxe_network: ${MAAS_PXE_NETWORK}
-    opnfv_fuel_maas_pxe_address: ${MAAS_PXE_NETWORK_ROOTSTR}.3
-    opnfv_fuel_maas_pxe_iprange_start: ${MAAS_PXE_NETWORK_ROOTSTR}.5
-    opnfv_fuel_maas_pxe_iprange_end: ${MAAS_PXE_NETWORK_ROOTSTR}.250
+    opnfv_maas_mcp_address: ${MAAS_IP}
+    opnfv_maas_pxe_network: ${MAAS_PXE_NETWORK}
+    opnfv_maas_pxe_address: ${MAAS_PXE_NETWORK_ROOTSTR}.3
+    opnfv_maas_pxe_iprange_start: ${MAAS_PXE_NETWORK_ROOTSTR}.5
+    opnfv_maas_pxe_iprange_end: ${MAAS_PXE_NETWORK_ROOTSTR}.250
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/init.yml
index 730193a..dc164dd 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/init.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/init.yml
@@ -20,7 +20,7 @@ parameters:
     infra_config_address: 10.167.4.100
     infra_config_deploy_address: 10.20.0.2
     infra_maas_node01_address: 10.167.4.3
-    infra_maas_node01_deploy_address: ${_param:opnfv_fuel_maas_pxe_address}
+    infra_maas_node01_deploy_address: ${_param:opnfv_maas_pxe_address}
     infra_maas_node01_external_address: 10.16.0.3
     infra_compute_node01_address: 10.167.4.141
     infra_compute_node02_address: 10.167.4.142
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/maas.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/maas.yml
index 72b9f52..c7006e7 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/maas.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-odl-ha/infra/maas.yml
@@ -32,13 +32,13 @@ parameters:
         network_discovery: 'enabled'
         default_min_hwe_kernel: 'hwe-16.04'
       subnets:
-        opnfv_fuel_maas_pxe:
-          name: ${_param:opnfv_fuel_maas_pxe_network}/24
-          cidr: ${_param:opnfv_fuel_maas_pxe_network}/24
+        opnfv_maas_pxe:
+          name: ${_param:opnfv_maas_pxe_network}/24
+          cidr: ${_param:opnfv_maas_pxe_network}/24
           gateway_ip: ${_param:single_address}
           iprange:
-            start: ${_param:opnfv_fuel_maas_pxe_iprange_start}
-            end: ${_param:opnfv_fuel_maas_pxe_iprange_end}
+            start: ${_param:opnfv_maas_pxe_iprange_start}
+            end: ${_param:opnfv_maas_pxe_iprange_end}
             type: dynamic
           vlans:
             untagged:
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/init.yml
index 94ccc9e..60a0b67 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/init.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/init.yml
@@ -20,7 +20,7 @@ parameters:
     infra_config_address: 10.167.4.100
     infra_config_deploy_address: 10.20.0.2
     infra_maas_node01_address: 10.167.4.3
-    infra_maas_node01_deploy_address: ${_param:opnfv_fuel_maas_pxe_address}
+    infra_maas_node01_deploy_address: ${_param:opnfv_maas_pxe_address}
     infra_maas_node01_external_address: 10.16.0.3
     infra_compute_node01_address: 10.167.4.141
     infra_compute_node02_address: 10.167.4.142
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/maas.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/maas.yml
index bdf7fea..61f35f5 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/maas.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-dpdk-ha/infra/maas.yml
@@ -32,13 +32,13 @@ parameters:
         network_discovery: 'enabled'
         default_min_hwe_kernel: 'hwe-16.04'
       subnets:
-        opnfv_fuel_maas_pxe:
-          name: ${_param:opnfv_fuel_maas_pxe_network}/24
-          cidr: ${_param:opnfv_fuel_maas_pxe_network}/24
+        opnfv_maas_pxe:
+          name: ${_param:opnfv_maas_pxe_network}/24
+          cidr: ${_param:opnfv_maas_pxe_network}/24
           gateway_ip: ${_param:single_address}
           iprange:
-            start: ${_param:opnfv_fuel_maas_pxe_iprange_start}
-            end: ${_param:opnfv_fuel_maas_pxe_iprange_end}
+            start: ${_param:opnfv_maas_pxe_iprange_start}
+            end: ${_param:opnfv_maas_pxe_iprange_end}
             type: dynamic
           vlans:
             untagged:
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml
index a75b41d..4520621 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml
@@ -24,7 +24,6 @@ parameters:
     reclass_data_repository: local
     salt_master_environment_repository: "https://github.com/tcpcloud"
     salt_master_environment_revision: master
-    reclass_config_master: ${_param:infra_config_deploy_address}
     single_address: ${_param:infra_config_address}
     deploy_address: ${_param:infra_config_deploy_address}
     salt_master_host: ${_param:infra_config_deploy_address}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml
index 6bf0fbe..64473fc 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml
@@ -17,19 +17,19 @@ parameters:
     infra_maas_database_password: opnfv_secret

     # infra service addresses
-    infra_config_address: 10.167.4.100
-    infra_config_deploy_address: 10.20.0.2
-    infra_maas_node01_address: 10.167.4.3
-    infra_maas_node01_deploy_address: ${_param:opnfv_fuel_maas_pxe_address}
-    infra_maas_node01_external_address: 10.16.0.3
-    infra_compute_node01_address: 10.167.4.141
-    infra_compute_node02_address: 10.167.4.142
-    infra_compute_node03_address: 10.167.4.143
+    infra_config_address: ${_param:opnfv_infra_config_address}
+    infra_config_deploy_address: ${_param:reclass_config_master}
+    infra_maas_node01_address: ${_param:opnfv_infra_maas_node01_address}
+    infra_maas_node01_deploy_address: ${_param:opnfv_maas_pxe_address}
+    infra_maas_node01_external_address: ${_param:opnfv_infra_maas_node01_external_address}
+    infra_compute_node01_address: ${_param:opnfv_infra_compute_node01_address}
+    infra_compute_node02_address: ${_param:opnfv_infra_compute_node02_address}
+    infra_compute_node03_address: ${_param:opnfv_infra_compute_node03_address}

-    infra_kvm_address: 10.167.4.140
-    infra_kvm_node01_address: 10.167.4.141
-    infra_kvm_node02_address: 10.167.4.142
-    infra_kvm_node03_address: 10.167.4.143
+    infra_kvm_address: ${_param:opnfv_infra_kvm_address}
+    infra_kvm_node01_address: ${_param:opnfv_infra_kvm_node01_address}
+    infra_kvm_node02_address: ${_param:opnfv_infra_kvm_node02_address}
+    infra_kvm_node03_address: ${_param:opnfv_infra_kvm_node03_address}

     infra_maas_node01_hostname: mas01
     infra_kvm_node01_hostname: kvm01
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml
index 7550c1a..5064cff 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml
@@ -143,17 +143,16 @@ parameters:
           enabled: true
           proto: dhcp
           type: bridge
-          name_servers:
-          - 8.8.8.8
-          - 8.8.4.4
+          name_servers: ${_param:opnfv_name_servers}
           use_interfaces:
           - ${_param:deploy_nic}
           noifupdown: true
-        vlan300:
+        mgmt-vlan:
+          name: vlan${_param:opnfv_net_mgmt_vlan}
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:deploy_nic}.300
+          name: ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
           use_interfaces:
           - ${_param:deploy_nic}
         br-ctl:
@@ -163,4 +162,4 @@ parameters:
           address: ${_param:single_address}
           netmask: 255.255.255.0
           use_interfaces:
-          - ${_param:deploy_nic}.300
+          - ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/maas.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/maas.yml
index f77abea..cb7709c 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/maas.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/maas.yml
@@ -15,7 +15,7 @@ parameters:
     maas_admin_username: opnfv
     maas_admin_password: opnfv_secret
     maas_db_password: opnfv_secret
-    dns_server01: 8.8.4.4
+    dns_server01: ${_param:opnfv_name_servers}
     single_address: ${_param:infra_maas_node01_deploy_address}
   maas:
     region:
@@ -32,13 +32,13 @@ parameters:
         network_discovery: 'enabled'
         default_min_hwe_kernel: 'hwe-16.04'
       subnets:
-        opnfv_fuel_maas_pxe:
-          name: ${_param:opnfv_fuel_maas_pxe_network}/24
-          cidr: ${_param:opnfv_fuel_maas_pxe_network}/24
+        opnfv_maas_pxe:
+          name: ${_param:opnfv_maas_pxe_network}/24
+          cidr: ${_param:opnfv_maas_pxe_network}/24
           gateway_ip: ${_param:single_address}
           iprange:
-            start: ${_param:opnfv_fuel_maas_pxe_iprange_start}
-            end: ${_param:opnfv_fuel_maas_pxe_iprange_end}
+            start: ${_param:opnfv_maas_pxe_iprange_start}
+            end: ${_param:opnfv_maas_pxe_iprange_end}
             type: dynamic
           vlans:
             untagged:
@@ -48,57 +48,57 @@ parameters:
       machines:
         kvm01:
           interface:
-            mac: "00:25:b5:a0:00:2a"
+            mac: ${_param:opnfv_maas_node01_interface_mac}
           power_parameters:
-            power_address: "172.30.8.75"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node01_power_address}
+            power_password: ${_param:opnfv_maas_node01_power_password}
+            power_type: ${_param:opnfv_maas_node01_power_type}
+            power_user: ${_param:opnfv_maas_node01_power_user}
+          architecture: ${_param:opnfv_maas_node01_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
         kvm02:
           interface:
-            mac: "00:25:b5:a0:00:3a"
+            mac: ${_param:opnfv_maas_node02_interface_mac}
           power_parameters:
-            power_address: "172.30.8.65"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node02_power_address}
+            power_password: ${_param:opnfv_maas_node02_power_password}
+            power_type: ${_param:opnfv_maas_node02_power_type}
+            power_user: ${_param:opnfv_maas_node02_power_user}
+          architecture: ${_param:opnfv_maas_node02_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
         kvm03:
           interface:
-            mac: "00:25:b5:a0:00:4a"
+            mac: ${_param:opnfv_maas_node03_interface_mac}
           power_parameters:
-            power_address: "172.30.8.74"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node03_power_address}
+            power_password: ${_param:opnfv_maas_node03_power_password}
+            power_type: ${_param:opnfv_maas_node03_power_type}
+            power_user: ${_param:opnfv_maas_node03_power_user}
+          architecture: ${_param:opnfv_maas_node03_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
         cmp001:
           interface:
-            mac: "00:25:b5:a0:00:5a"
+            mac: ${_param:opnfv_maas_node04_interface_mac}
           power_parameters:
-            power_address: "172.30.8.73"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node04_power_address}
+            power_password: ${_param:opnfv_maas_node04_power_password}
+            power_type: ${_param:opnfv_maas_node04_power_type}
+            power_user: ${_param:opnfv_maas_node04_power_user}
+          architecture: ${_param:opnfv_maas_node04_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
         cmp002:
           interface:
-            mac: "00:25:b5:a0:00:6a"
+            mac: ${_param:opnfv_maas_node05_interface_mac}
           power_parameters:
-            power_address: "172.30.8.72"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node05_power_address}
+            power_password: ${_param:opnfv_maas_node05_power_password}
+            power_type: ${_param:opnfv_maas_node05_power_type}
+            power_user: ${_param:opnfv_maas_node05_power_user}
+          architecture: ${_param:opnfv_maas_node05_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
   linux:
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml
index 93a26b1..a48d924 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml
@@ -67,19 +67,19 @@ parameters:
           netmask: 255.255.255.0
           mtu: 1500
           use_interfaces:
-          - ${_param:tenant_nic}.302
-        vlan300:
+          - ${_param:tenant_nic}.${_param:opnfv_net_tenant_vlan}
+        mgmt-vlan:
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:mgmt_nic}.300
+          name: ${_param:mgmt_nic}.${_param:opnfv_net_mgmt_vlan}
           use_interfaces:
           - ${_param:mgmt_nic}
-        vlan302:
+        tenant-vlan:
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:tenant_nic}.302
+          name: ${_param:tenant_nic}.${_param:opnfv_net_tenant_vlan}
           use_interfaces:
           - ${_param:tenant_nic}
         br-ctl:
@@ -89,7 +89,7 @@ parameters:
           address: ${_param:single_address}
           netmask: 255.255.255.0
           use_interfaces:
-          - ${_param:mgmt_nic}.300
+          - ${_param:mgmt_nic}.${_param:opnfv_net_mgmt_vlan}
         br-floating:
           enabled: true
           type: ovs_bridge
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml
index 9c2bc7b..eff5651 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml
@@ -3,32 +3,32 @@ parameters:

     openstack_version: ocata

-    openstack_gateway_node01_address: 10.167.4.124
-    openstack_gateway_node02_address: 10.167.4.125
-    openstack_gateway_node03_address: 10.167.4.126
-    openstack_gateway_node01_tenant_address: 10.1.0.6
-    openstack_gateway_node02_tenant_address: 10.1.0.7
-    openstack_gateway_node03_tenant_address: 10.1.0.9
+    openstack_gateway_node01_address: ${_param:opnfv_openstack_gateway_node01_address}
+    openstack_gateway_node02_address: ${_param:opnfv_openstack_gateway_node02_address}
+    openstack_gateway_node03_address: ${_param:opnfv_openstack_gateway_node03_address}
+    openstack_gateway_node01_tenant_address: ${_param:opnfv_openstack_gateway_node01_tenant_address}
+    openstack_gateway_node02_tenant_address: ${_param:opnfv_openstack_gateway_node02_tenant_address}
+    openstack_gateway_node03_tenant_address: ${_param:opnfv_openstack_gateway_node03_tenant_address}
     openstack_gateway_node01_hostname: gtw01
     openstack_gateway_node02_hostname: gtw02
     openstack_gateway_node03_hostname: gtw03

     # openstack service addresses
-    openstack_proxy_address: 10.167.4.80
-    openstack_proxy_node01_address: 10.167.4.81
-    openstack_proxy_node02_address: 10.167.4.82
-    openstack_control_address: 10.167.4.10
-    openstack_control_node01_address: 10.167.4.11
-    openstack_control_node02_address: 10.167.4.12
-    openstack_control_node03_address: 10.167.4.13
-    openstack_database_address: 10.167.4.50
-    openstack_database_node01_address: 10.167.4.51
-    openstack_database_node02_address: 10.167.4.52
-    openstack_database_node03_address: 10.167.4.53
-    openstack_message_queue_address: 10.167.4.40
-    openstack_message_queue_node01_address: 10.167.4.41
-    openstack_message_queue_node02_address: 10.167.4.42
-    openstack_message_queue_node03_address: 10.167.4.43
+    openstack_proxy_address: ${_param:opnfv_openstack_proxy_address}
+    openstack_proxy_node01_address: ${_param:opnfv_openstack_proxy_node01_address}
+    openstack_proxy_node02_address: ${_param:opnfv_openstack_proxy_node02_address}
+    openstack_control_address: ${_param:opnfv_openstack_control_address}
+    openstack_control_node01_address: ${_param:opnfv_openstack_control_node01_address}
+    openstack_control_node02_address: ${_param:opnfv_openstack_control_node02_address}
+    openstack_control_node03_address: ${_param:opnfv_openstack_control_node03_address}
+    openstack_database_address: ${_param:opnfv_openstack_database_address}
+    openstack_database_node01_address: ${_param:opnfv_openstack_database_node01_address}
+    openstack_database_node02_address: ${_param:opnfv_openstack_database_node02_address}
+    openstack_database_node03_address: ${_param:opnfv_openstack_database_node03_address}
+    openstack_message_queue_address: ${_param:opnfv_openstack_message_queue_address}
+    openstack_message_queue_node01_address: ${_param:opnfv_openstack_message_queue_node01_address}
+    openstack_message_queue_node02_address: ${_param:opnfv_openstack_message_queue_node02_address}
+    openstack_message_queue_node03_address: ${_param:opnfv_openstack_message_queue_node03_address}


     openstack_telemetry_hostname: mdb
@@ -36,23 +36,23 @@ parameters:
     openstack_telemetry_node02_hostname: mdb02
     openstack_telemetry_node03_hostname: mdb03

-    openstack_telemetry_address: 10.167.4.75
-    openstack_telemetry_node01_address: 10.167.4.76
-    openstack_telemetry_node02_address: 10.167.4.77
-    openstack_telemetry_node03_address: 10.167.4.78
+    openstack_telemetry_address: ${_param:opnfv_openstack_telemetry_address}
+    openstack_telemetry_node01_address: ${_param:opnfv_openstack_telemetry_node01_address}
+    openstack_telemetry_node02_address: ${_param:opnfv_openstack_telemetry_node02_address}
+    openstack_telemetry_node03_address: ${_param:opnfv_openstack_telemetry_node03_address}

     # OpenStack Compute
-    openstack_compute_node01_single_address: 10.167.4.101
-    openstack_compute_node02_single_address: 10.167.4.102
-    openstack_compute_node03_single_address: 10.167.4.103
-    openstack_compute_node01_control_address: 10.167.4.101
-    openstack_compute_node02_control_address: 10.167.4.102
-    openstack_compute_node03_control_address: 10.167.4.103
-    openstack_compute_node01_tenant_address: 10.1.0.101
-    openstack_compute_node02_tenant_address: 10.1.0.102
-    openstack_compute_node03_tenant_address: 10.1.0.103
-    openstack_compute_node01_external_address: 172.30.10.2
-    openstack_compute_node02_external_address: 172.30.10.3
+    openstack_compute_node01_single_address: ${_param:opnfv_openstack_compute_node01_single_address}
+    openstack_compute_node02_single_address: ${_param:opnfv_openstack_compute_node02_single_address}
+    openstack_compute_node03_single_address: ${_param:opnfv_openstack_compute_node03_single_address}
+    openstack_compute_node01_control_address: ${_param:opnfv_openstack_compute_node01_control_address}
+    openstack_compute_node02_control_address: ${_param:opnfv_openstack_compute_node02_control_address}
+    openstack_compute_node03_control_address: ${_param:opnfv_openstack_compute_node03_control_address}
+    openstack_compute_node01_tenant_address: ${_param:opnfv_openstack_compute_node01_tenant_address}
+    openstack_compute_node02_tenant_address: ${_param:opnfv_openstack_compute_node02_tenant_address}
+    openstack_compute_node03_tenant_address: ${_param:opnfv_openstack_compute_node03_tenant_address}
+    openstack_compute_node01_external_address: ${_param:opnfv_openstack_compute_node01_external_address}
+    openstack_compute_node02_external_address: ${_param:opnfv_openstack_compute_node02_external_address}

     # openstack service hostnames
     openstack_proxy_hostname: prx
